File: InternetForumFrontend/src/app/Models/UserPostCommentVote.ts
import { PostCommentVote } from "./PostCommentVote";
import { User } from "./User";

// NOTE: ? = null, ! = not null
export class UserPostCommentVote {
        userId?: number;
        user?: User;
        postCommentVote?: PostCommentVote;
        postCommentVoteId?: number;
        postCommentVoteDirection!: number;
}

File: InternetForumBackend/API/Data/Models/UserPostCommentVote.cs
namespace API.Data.Models
{
    public class UserPostCommentVote
    {
        public int UserId { get; set; }
        public User User { get; set; }

        public int PostCommentVoteId { get; set; }

        public PostCommentVote PostCommentVote { get; set; }

        //upvote/downvote
        public int PostCommentVoteDirection { get; set; }
    }
}
